src/js/utils/modalWindow.js

```diff
@@ -0,0 +1,77 @@
+export class ModalWindow {
+  onOpenCallback = null;
+  onCloseCallback = null;
+  modalContents = '';
+
+  modalContentElement = document.querySelector('.modal__content');
+  closeButtonElement = document.querySelector('.modal__close-btn');
+  backdropElement = document.querySelector('.modal__backdrop');
+
+  onKeyPressBound = this.onKeyPress.bind(this);
+  onCloseBound = this.onMouseClick.bind(this);
+
+  constructor(modalRootElement, options = {}) {
+    this.backdropElement = modalRootElement;
+
+    for (const [key, value] of Object.entries(options)) this[key] = value;
+  }
+
+  show(modalContents) {
+    if ('' === modalContents) return this.close();
+    if (!'' === this.modalContents) return this.update(modalContents);
```

Comment:
does this condition work?
`!''` is always true
`this.modalContents` string?

Reply:
This is **yoda notation**, same thing as `(!this.modalContents === '')`. Protects from inadvertent assignments like this: `(!this.modalContents = '')`